include/openPMD/RecordComponent.hpp

```diff
@@ -225,29 +197,118 @@ class RecordComponent : public BaseRecordComponent
+     *               Optimizations might be implemented based on this
+     *               assumption (e.g. further deferring the operation if the
+     *               backend is the unique owner).
+     *               For raw pointers, use storeChunkRaw().
+     * @param offset Offset within the dataset.
+     * @param extent Extent within the dataset, counted from the offset.
      */
     template <typename T>
-    void loadChunk(std::shared_ptr<T>, Offset, Extent);
+    void storeChunk(std::shared_ptr<T> data, Offset offset, Extent extent);
 
+    /** Store a chunk of data from a chunk of memory, array version.
+     *
+     * @param data   Preallocated, contiguous buffer, large enough to read the
+     *               the specified data from it.
+     *               The array-based overload helps avoid having to manually
+     *               specify the delete[] destructor (C++17 feature).
+     * @param offset Offset within the dataset.
+     * @param extent Extent within the dataset, counted from the offset.
+     */
     template <typename T>
-    void storeChunk(std::shared_ptr<T>, Offset, Extent);
+    void storeChunk(std::shared_ptr<T[]> data, Offset offset, Extent extent);
 
+    /** Store a chunk of data from a chunk of memory, raw pointer version.
+     *
+     * @param data   Preallocated, contiguous buffer, large enough to read the
+     *               the specified data from it.
+     *               It is in the user of this API call's responsibility to
+     *               ensure that the lifetime of the buffer exceeds the next
+     *               <a
+     * href="https://openpmd-api.readthedocs.io/en/latest/usage/workflow.html#deferred-data-api-contract">
+     *               flush point</a>.
+     * @param offset Offset within the dataset.
+     * @param extent Extent within the dataset, counted from the offset.
+     */
     template <typename T>
-    void storeChunk(std::shared_ptr<T[]>, Offset, Extent);
+    void storeChunkRaw(T *data, Offset offset, Extent extent);
 
+    /** Store a chunk of data from a contiguous container.
+     *
+     * @param data   <a
+     *               href="https://en.cppreference.com/w/cpp/named_req/ContiguousContainer">
+     *               Contiguous container</a>, large enough to read the the
+     *               specified data from it. A contiguous container in here is
+     *               either a std::vector or a std::array.
+     *               It is in the user of this API call's responsibility to
+     *               ensure that the lifetime of the container exceeds the next
+     *               <a
+     * href="https://openpmd-api.readthedocs.io/en/latest/usage/workflow.html#deferred-data-api-contract">
+     *               flush point</a>.
+     * @param offset Offset within the dataset.
+     * @param extent Extent within the dataset, counted from the offset.
+     */
     template <typename T_ContiguousContainer>
     typename std::enable_if<
```

Comment:
you can use `enable_if_t` :)

Reply:
This is old code, just moved around. But I'll update it